lib/rpc/rpc_mainchain.dart

```diff
@@ -36,4 +86,31 @@ class MainchainRPCLive implements MainchainRPC {
     const kbyteInTx = 5;
     return btcPerKb * kbyteInTx;
   }
+
+  @override
+  Future<(bool, String?)> testConnection() async {
+    try {
+      final pong = await _client?.call('ping') as Map<String, dynamic>?;
+      if (pong == null) {
+        connectionError = '';
+        connected = true;
+      } else {
+        connectionError = pong['error'];
```

Comment:
`ping` doesn't return anything at all, it looks like. Instead of checking for an `error` key, you must catch any potential exceptions being thrown. They are of `RPCException` type, I believe

Reply:
<img width="682" alt="image" src="https://github.com/barebitcoin/sidesail/assets/16610775/4f600286-f5eb-4999-8dd1-5221bc977a35">

Reply:
This is from a successfull ping, FYI

Reply:
do we send jsonrpc requests though?

Reply:
Yes, this is what happens behind the curtains of our RPC lib